File: interview questions/mergeSort.py
import math


def mergeSort(array):
    if len(array) <= 1:
        return array

       mid = math.floor(len(array) / 2)
        left = array[0:mid]
        right = array[mid:]

        sortedLeft = mergeSort(left)
        sortedRight = mergeSort(right)

        return merge(sortedLeft, sortedRight)


def merge(left, right):
    sortedArray = []

    while (len(left) > 0 and len(right) > 0):
        if left[0] < right[0]:
            sortedArray.append(left.pop(0))
        else:
            sortedArray.append(right.pop(0))
    return sortedArray + left + right

File: interview questions/mergeSort.js
function mergeSort(array) {
  if (array.length <= 1) return array;

  let mid = Math.floor(array.length / 2);
  let left = array.slice(0, mid);
  let right = array.slice(mid);

  let sortedLeft = mergeSort(left);
  let sortedRight = mergeSort(right);

  return merge(sortedLeft, sortedRight);
}

function merge(left, right) {
  let sortedArray = [];

  while (left.length > 0 && right.length > 0) {
    if (left[0] < right[0]) {
      sortedArray.push(left.shift());
    } else {
      sortedArray.push(right.shift());
    }
  }

  return [...sortedArray, ...left, ...right];
}
